### PR TITLE
feat: Add experimental metrics logging middleware

### PR DESCRIPTION
**Pull request by nand4011**
Add abstract middleware that captures metrics for each grpc request.

Add metrics middleware implementation that writes the metrics to a CSV.

Add metrics middleware implementation that logs the metrics.

### CODE REVIEW

**Comment by nand4011** [comment]
Compare to the Typescript metrics middleware added in this pr: https://github.com/momentohq/client-sdk-javascript/pull/675

File: src/Momento.Sdk/Config/Middleware/ExperimentalMetricsCsvMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        }
+
+        /// <summary>
+        /// Writes metrics for a request out to a CSV.
+        /// </summary>
+        public override Task EmitMetrics(ExperimentalRequestMetrics metrics)
+        {
+            var csvLine = $"{metrics.NumActiveRequestsAtStart}, {metrics.NumActiveRequestsAtFinish}, " +
+                                     $"{metrics.RequestType}, {metrics.Status}, {metrics.StartTime}, {metrics.EndTime}," +
+                                     $"{metrics.Duration}, {metrics.RequestSize}, {metrics.ResponseSize}";
```
Comment (malandis) [inline src/Momento.Sdk/Config/Middleware/ExperimentalMetricsCsvMiddleware.cs:50]: Is there a standard CSV writer in .NET?

File: src/Momento.Sdk/Config/Middleware/ExperimentalMetricsCsvMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+            lock (_lock)
+            {
+                _writer.WriteLine(csvLine);
+                _writer.Flush();
+            }
```
Comment (malandis) [inline src/Momento.Sdk/Config/Middleware/ExperimentalMetricsCsvMiddleware.cs:56]: 😎

File: src/Momento.Sdk/Config/Middleware/ExperimentalMetricsMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    /// <summary>
+    /// A struct that records metrics about a Momento request. These metrics include the number of requests active 
+    /// at the start and finish of the request, the type of the request, the status code of the response, 
+    /// timing information about the request (start, end, and duration times in milliseconds), 
+    /// and the size of the request and response bodies in bytes.
+    ///
+    /// Metrics may be added to this struct in the future if they are useful.
+    /// </summary>
+    public record struct ExperimentalRequestMetrics(
```
Comment (malandis) [inline src/Momento.Sdk/Config/Middleware/ExperimentalMetricsMiddleware.cs:18]: 👍

**Review by malandis — APPROVED** [review]

File: src/Momento.Sdk/Config/Middleware/ExperimentalMetricsCsvMiddleware.cs (hunk 1 above)
Comment (nand4011) [reply src/Momento.Sdk/Config/Middleware/ExperimentalMetricsCsvMiddleware.cs:50]: I didn't see one. I figured this was simple enough that I could get away with manually putting the string together.

**Pull request merged**